src/viur/datastore/types.py

```diff
@@ -164,6 +164,13 @@ def __init__(self, key: Optional[Key] = None, exclude_from_indexes: Optional[Set
 		assert isinstance(self.exclude_from_indexes, set)
 		self.version = None
 
+	@property
+	def exclude_from_indexes(self):
+		return self._exclude_from_indexes
+
+	@exclude_from_indexes.setter
+	def exclude_from_indexes(self, value):
```

Comment:
Okay, that's another way. 
I would have forced the type via type check:
```py
		if not isinstance(value, set):
			raise TypeError("exclude_from_indexes must be a set")
```
But you can also do it this way. It must now only be clear that a copy is ALWAYS created. I.e.
```py
my_excluded_keys =  {"foo"}
entity.exclude_from_indexes = my_excluded_keys
my_excluded_keys.add("bar")
```
no longer works. But for me this is fine

--- 
But please add some type hints:


```suggestion
	@exclude_from_indexes.setter
	def exclude_from_indexes(self, value : set[str] | list[str] | tuple[str]) -> None:
```

Reply:
> Okay, that's another way. I would have forced the type via type check:
> 
> ```python
> 		if not isinstance(value, set):
> 			raise TypeError("exclude_from_indexes must be a set")
> ```
> 
> But you can also do it this way. It must now only be clear that a copy is ALWAYS created. I.e.
> 
> ```python
> my_excluded_keys =  {"foo"}
> entity.exclude_from_indexes = my_excluded_keys
> my_excluded_keys.add("bar")
> ```
> 
> no longer works. But for me this is fine


```python
exclude_indexes = ["foo"]
key = db.Key("db-test", "my-entity")
entity = db.Entity(key)
entity["foo"] = "x"
entity["bar"] = "y"
entity.exclude_from_indexes = exclude_indexes
entity.exclude_from_indexes.add("bar")
db.Put(entity)
print(f"{entity = }")
print(f"{entity.exclude_from_indexes = }")
entity = db.Get(key)
print(f"{entity = }")
print(f"{entity.exclude_from_indexes = }")
```
```
entity = {'foo': 'x', 'bar': 'y'}
entity.exclude_from_indexes = {'bar', 'foo'}
entity = {'bar': 'y', 'foo': 'x'}
entity.exclude_from_indexes = {'bar', 'foo'}
```
This works for me